flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/ExecutionAttemptMappingProvider.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.checkpoint;
+
+import org.apache.flink.runtime.executiongraph.ExecutionAttemptID;
+import org.apache.flink.runtime.executiongraph.ExecutionVertex;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.util.Preconditions.checkNotNull;
+
+/**
+ * Responses to query the current attempt for the tasks and provide the mapping from the execution
+ * attempt id to its task.
+ */
+public class ExecutionAttemptMappingProvider {
+
+    private final List<ExecutionVertex> tasks;
```

Comment:
This works because the list of `ExecutionVertex` is currently static in the `ExecutionGraph` for the lifetime of the execution graph, I hope.

It might make sense to turn this into an interface that `ExecutionGraph` implements to allow querying this information. It's just an idea, so please feel free to ignore for now.

Reply:
I think the concern does make sense since we are considering reactive scheduling right now. I think we might first keep the static list in this PR since a lot of checkpoint related tests are relying on mocked execution vertex right now and we would change them to real `ExecutionGraph` in the next PR, otherwise we might need to also modify the tests in this PR. And if we consider dynamic  `ExecutionGraph` we might also need an end-to-end modification to not 
depend on the number of tasks in initializing the cached mapping (e.g., in `removeEldestEntry()`).